src/test/java/uk/gov/homeoffice/digital/sas/timecard/listeners/KafkaEntityListenerIT.java

```diff
@@ -0,0 +1,172 @@
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.transaction.support.TransactionSynchronizationManager;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.read.ListAppender;
+import uk.gov.homeoffice.digital.sas.timecard.kafka.KafkaDbTransactionSynchronizer;
+import uk.gov.homeoffice.digital.sas.timecard.model.TimeEntry;
+import uk.gov.homeoffice.digital.sas.timecard.testutils.CommonUtils;
+import uk.gov.homeoffice.digital.sas.timecard.testutils.TimeEntryFactory;
+
+@SpringBootTest
+@DirtiesContext
+@WebAppConfiguration
+@AutoConfigureMockMvc(addFilters = true)
+@EmbeddedKafka(partitions = 1, brokerProperties = { "listeners=PLAINTEXT://localhost:9092",
+    "port=9092" })
+class KafkaEntityListenerIT<T> {
+
+  @Autowired
+  private MockMvc mockMvc;
+
+  private TimeEntry timeEntry;
+
+  private UUID tenantId;
+
+  private UUID ownerId;
+
+  private String messageKey;
+
+  @BeforeEach
+  void setUp() {
+    ownerId = UUID.randomUUID();
+    timeEntry = TimeEntryFactory.createTimeEntry(ownerId,
+        CommonUtils.getAsDate(LocalDateTime.now()));
+    tenantId = timeEntry.getTenantId();
+    messageKey = CommonUtils.generateMessageKey(timeEntry);
+    TransactionSynchronizationManager.initSynchronization();
+  }
+
+  @Test
+  void givenValidRequest_WhenSendingCreateRequest_thenTransactionSyncLogsSuccessMessage()
+      throws Exception {
+    ListAppender<ILoggingEvent> listAppender = getLoggingEventListAppender();
+
+    List<ILoggingEvent> logList = listAppender.list;
+
+    persistTimeEntry(timeEntry);
+
+    assertEquals(String.format(
+        "Kafka Transaction [ create ] Initialized with message key [ %s ]",
+        messageKey), logList.get(0).getMessage());
+
+    assertEquals("Database transaction [ create ] was successful", logList.get(1).getMessage());
+
+    assertEquals(String.format(
+        "Transaction successful with messageKey [ %s ]", messageKey), logList.get(2).getMessage());
+  }
+
+  @Test
+  void givenValidRequest_WhenSendingUpdateRequest_thenTransactionSyncLogsSuccessMessage()
+    throws Exception {
+    ListAppender<ILoggingEvent> listAppender = getLoggingEventListAppender();
+
+    List<ILoggingEvent> logList = listAppender.list;
+
+    String id = persistTimeEntry(timeEntry);
+
+    TimeEntry timeEntryUpdate = TimeEntryFactory.createTimeEntry(ownerId, tenantId,
+        CommonUtils.getAsDate(LocalDateTime.now().plusHours(1)));
+
+    mockMvc.perform(put("/resources/time-entries/"+ id + "?tenantId=" + tenantId)
+          .contentType(MediaType.APPLICATION_JSON)
+          .content(CommonUtils.timeEntryAsJsonString(timeEntryUpdate)))
+        .andDo(print())
+        .andExpect(status().isOk());
+
+    assertEquals(String.format(
```

Comment:
It might be worth while to assert that the logs before index 3 **do not** contain any db transaction log to give us certainty that it definitely hasn't occurred before our kafka log at index 3
And maybe consider this for the other tests here too if applicable

Reply:
I have add in 
`    List<ILoggingEvent> filteredList =
        logList.stream().filter(o -> o.getMessage().equals(
            String.format(
        "Database transaction [ update ] with entity id [ %s ] was successful", id))).toList();

    assertThat(filteredList).hasSize(1);`
    
   Ensuring we only have on occurrence of that message.

Reply:
looks good :+1: